advisor-go/src/quixo/game/game.go

```diff
@@ -1,2 +1,167 @@
+package game
 
+import (
+	"fmt"
+	"strconv"
+)
 
+// Board with Grid and Player
+type Board struct {
+	Grid   [][]int
+	Player int
+}
+
+// Coords x and y
+type Coords struct {
+	X int
+	Y int
+}
+
+// Cube with Coords and value
+type Cube struct {
+	Coords Coords
+	Value  int
+}
+
+// GetBoard return a Board from grid and player
+func GetBoard(grid [][]int, player int) Board {
+	newBoard := Board{
+		Grid:   grid,
+		Player: player,
+	}
+	return newBoard
+}
+
+// GetMovablesCubes return an array of movables cube
+func GetMovablesCubes(board Board) []Cube {
+	grid := board.Grid
+	player := board.Player
+
+	movables := []Cube{}
+	for x := 0; x < len(grid); x++ {
+		for y := 0; y < len(grid[x]); y++ {
+			cube := Cube{
+				Coords: Coords{X: x, Y: y},
+				Value:  grid[x][y],
+			}
+			if isOutsideCube(cube.Coords, len(grid[x])) && isPlayerCube(cube.Value, player) {
+				movables = append(movables, cube)
+			}
+		}
+	}
+	return movables
+}
+
+// GetAvailablesDestinations return an array of destinations
+func GetAvailablesDestinations(grid [][]int, coords Coords) []Coords {
+	destinations := []Coords{}
+	x := coords.X
+	y := coords.Y
+	indexEnd := len(grid) - 1
+
+	if x == 0 || x == indexEnd {
```

Comment:
this part is hard to follow.

Reply:
Added commentary and lambda to make it more clearer